lib/opensearch/api/actions/asynchronous_search/delete.rb

```diff
@@ -0,0 +1,38 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+# This file is generated from the OpenSearch REST API spec.
+# Do not modify it by hand. Instead, modify the generator or the spec.
+
+# frozen_string_literal: true
+
+module OpenSearch
+  module API
+    module AsynchronousSearch
+      module Actions
+        # Delete asynchronous search.
+        #
+        # @option args [String] :id *Required*
+        # @option args [List] :ignore set to [404] to ignore server's NOT FOUND error for this request
+        def delete(args = {})
+          args = Utils.normalize_arguments(args)
+          raise ArgumentError, "Required argument 'id' missing" unless args['id']
+
+          _id = args.delete('id')
```

Comment:
If `normalize_arguments` always dups them we can rename it to `normalize_and_dup_arguments`, but I'd prefer if we explicitly dup'ed them.

Reply:
The process of normalizing already creates a new hash. I'll rename it for clarity.